src/rgw/rgw_rados.cc

```diff
@@ -8780,6 +8778,7 @@ int RGWRados::cls_bucket_list_unordered(const DoutPrefixProvider *dpp,
 	  ent_list.emplace_back(std::move(dirent));
 	  ++count;
 	} else {
+	  last_added_entry = dirent.key;
```

Comment:
I believe this alone is the solution, right?

Reply:
Absolutely right！